### PR TITLE
Add support for Consumer Create Action

### PR DESCRIPTION
**Pull request by Ivandemidov00**
Resolves [#475](https://github.com/nats-io/nats.net.v2/issues/475)

### CODE REVIEW

**Comment by Ivandemidov00** [comment]
Fixed problem with serialize .net8 and test on create publisher action

File: src/NATS.Client.JetStream/Models/ConsumerCreateRequest.cs
Comment (mtmk) [inline src/NATS.Client.JetStream/Models/ConsumerCreateRequest.cs:39]: Could we make this optional so that we don't have to deal with net8 serialization exception for an empty value?

File: src/NATS.Client.JetStream/Models/ConsumerCreateRequestAction.cs
Comment (mtmk) [inline src/NATS.Client.JetStream/Models/ConsumerCreateRequestAction.cs:10]: we can remove this option if action is optional.

File: src/NATS.Client.JetStream/Internal/NatsJSJsonSerializer.cs
Comment (mtmk) [inline src/NATS.Client.JetStream/Internal/NatsJSJsonSerializer.cs:350]: is this for net8? I guess we can leave it as camelCase and not need this class?

File: src/NATS.Client.JetStream/NatsJSContext.Consumers.cs
Comment (mtmk) [inline src/NATS.Client.JetStream/NatsJSContext.Consumers.cs:36]: we could make action nullable here.

File: src/NATS.Client.JetStream/Models/ConsumerCreateRequestAction.cs
Comment (mtmk) [inline src/NATS.Client.JetStream/Models/ConsumerCreateRequestAction.cs:5]: what do you think of `ConsumerCreateAction` as the name? more succinct?

**Review by mtmk — COMMENTED** [review]
looks good over all, just a suggestion about removing CreateOrUpdate from the enum since absence of the action field would mean the same thing as an empty string.

File: src/NATS.Client.JetStream/Models/ConsumerCreateRequestAction.cs
Comment (Ivandemidov00) [reply src/NATS.Client.JetStream/Models/ConsumerCreateRequestAction.cs]: Yes, think it's good too. Renamed

File: src/NATS.Client.JetStream/Models/ConsumerCreateRequest.cs
Comment (Ivandemidov00) [reply src/NATS.Client.JetStream/Models/ConsumerCreateRequest.cs]: Yes, that was one of the ideas. In this case, the property will sample be ignored and `NatsJSJsonStringEnumConverter` will be less strict

**Comment by mtmk** [comment]
@Ivandemidov00 thanks for the updates. Just made a change to enum default and removed the nullable marking. I think it's easier to reason with this way.

**Review by mtmk — APPROVED** [review]
LGTM thanks @Ivandemidov00 
(will wait for another review because of my changes)

File: src/NATS.Client.JetStream/Models/ConsumerCreateAction.cs
Comment (caleblloyd) [inline src/NATS.Client.JetStream/Models/ConsumerCreateAction.cs:5]: Would it make sense to make the default descriptive?  I.e. `CreateOrUpdate = 0` instead of `Default = 0`?

Or is it best to leave it vague as the NATS Server can decide the default?

**Review by caleblloyd — APPROVED** [review]
LGTM

**Pull request merged**
